native/explorer/src/lazyframe.rs

```diff
@@ -0,0 +1,71 @@
+use polars::prelude::*;
+use std::result::Result;
+
+use crate::{ExDataFrame, ExLazyFrame, ExSeries, ExplorerError};
+
+#[rustler::nif(schedule = "DirtyCpu")]
+pub fn lf_collect(data: ExLazyFrame) -> Result<ExDataFrame, ExplorerError> {
+    Ok(ExDataFrame::new(data.resource.0.clone().collect()?))
+}
+
+#[rustler::nif(schedule = "DirtyCpu")]
+pub fn lf_fetch(data: ExLazyFrame, n_rows: usize) -> Result<ExDataFrame, ExplorerError> {
+    Ok(ExDataFrame::new(data.resource.0.clone().fetch(n_rows)?))
+}
+
+#[rustler::nif(schedule = "DirtyCpu")]
+pub fn lf_describe_plan(data: ExLazyFrame, optimized: bool) -> Result<String, ExplorerError> {
+    let lf = &data.resource.0;
+    let plan = match optimized {
+        true => lf.describe_optimized_plan()?,
+        false => lf.describe_plan(),
+    };
+    Ok(plan)
+}
+
+#[rustler::nif(schedule = "DirtyCpu")]
+pub fn lf_head(data: ExLazyFrame, length: u32) -> Result<ExLazyFrame, ExplorerError> {
+    let lf = &data.resource.0;
+    Ok(ExLazyFrame::new(lf.clone().limit(length)))
+}
+
+#[rustler::nif(schedule = "DirtyCpu")]
+pub fn lf_tail(data: ExLazyFrame, length: u32) -> Result<ExLazyFrame, ExplorerError> {
+    let lf = &data.resource.0;
+    Ok(ExLazyFrame::new(lf.clone().tail(length)))
+}
+
+#[rustler::nif(schedule = "DirtyCpu")]
+pub fn lf_pull(data: ExLazyFrame, name: &str) -> Result<ExSeries, ExplorerError> {
+    let df = &data.resource.0.clone().select(&[col(name)]).collect()?;
+    let series = df.column(name).map(|s| ExSeries::new(s.clone()))?;
+    Ok(series)
+}
+
+#[rustler::nif(schedule = "DirtyCpu")]
+pub fn lf_names(data: ExLazyFrame) -> Result<Vec<String>, ExplorerError> {
+    let lf = &data.resource.0;
+    Ok(lf.schema().iter_names().cloned().collect())
+}
+
+#[rustler::nif(schedule = "DirtyCpu")]
+pub fn lf_dtypes(data: ExLazyFrame) -> Result<Vec<String>, ExplorerError> {
+    let lf = &data.resource.0;
+    Ok(lf
+        .schema()
+        .iter_dtypes()
+        .map(|dtype| dtype.to_string())
+        .collect())
+}
+
+#[rustler::nif(schedule = "DirtyCpu")]
+pub fn lf_select(data: ExLazyFrame, columns: Vec<&str>) -> Result<ExLazyFrame, ExplorerError> {
+    let lf = &data.resource.0.clone().select(&[cols(columns)]);
+    Ok(ExLazyFrame::new(lf.clone()))
+}
+
+#[rustler::nif(schedule = "DirtyCpu")]
+pub fn lf_drop(data: ExLazyFrame, columns: Vec<&str>) -> Result<ExLazyFrame, ExplorerError> {
+    let lf = &data.resource.0.clone().select(&[col("*").exclude(columns)]);
+    Ok(ExLazyFrame::new(lf.clone()))
+}
```

Comment:
Btw, if those operations are very fast (it seems to be the case), they don't need to be dirty nif. I would expect most lazy operations to be fast, except the collect ones. :)

I would also expect the names/dtypes operations in regular data frames to not need to be dirty. Anything akin to a "property read" should be fine.

Reply:
Good point -- that's a holdover from copy-paste :).